paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentSheetScreen.kt

```diff
@@ -335,18 +343,54 @@ private fun PrimaryButton(viewModel: BaseSheetViewModel, type: PaymentSheetFlowT
             }
         }
 
-    when (type) {
-        Complete -> {
-            AndroidViewBinding(
-                factory = StripeFragmentPaymentSheetPrimaryButtonBinding::inflate,
-                modifier = modifier,
+    var button by remember {
+        mutableStateOf<PrimaryButton?>(null)
+    }
+
+    val context = LocalContext.current
+
+    AndroidViewBinding(
+        factory = { inflater: LayoutInflater, parent: ViewGroup, attachToParent: Boolean ->
+            val binding = StripeFragmentPrimaryButtonContainerBinding.inflate(inflater, parent, attachToParent)
```

Comment:
This moves much of what was in a fragment before, to inline here. This makes it so there's no disconnect between the view model, which is what was missing and causing exceptions with "don't keep activities".